homeassistant/components/github/sensor.py

```diff
@@ -1,293 +1,355 @@
+    RepositoryReleaseDataUpdateCoordinator,
 )
 
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {
-        vol.Required(CONF_ACCESS_TOKEN): cv.string,
-        vol.Optional(CONF_URL): cv.url,
-        vol.Required(CONF_REPOS): vol.All(cv.ensure_list, [REPO_SCHEMA]),
-    }
+
+@dataclass
+class GitHubSensorBaseEntityDescriptionMixin:
+    """Mixin for required GitHub base description keys."""
+
+    coordinator_key: CoordinatorKeyType
+
+
+@dataclass
+class GitHubSensorInformationEntityDescriptionMixin(
+    GitHubSensorBaseEntityDescriptionMixin
+):
+    """Mixin for required GitHub information description keys."""
+
+    value_fn: Callable[[GitHubRepositoryModel], StateType]
+
+
+@dataclass
+class GitHubSensorIssueEntityDescriptionMixin(GitHubSensorBaseEntityDescriptionMixin):
+    """Mixin for required GitHub information description keys."""
+
+    value_fn: Callable[[IssuesPulls], StateType]
+
+
+@dataclass
+class GitHubSensorBaseEntityDescription(SensorEntityDescription):
+    """Describes GitHub sensor entity default overrides."""
+
+    icon: str = "mdi:github"
+    entity_registry_enabled_default: bool = False
+
+
+@dataclass
+class GitHubSensorInformationEntityDescription(
+    GitHubSensorBaseEntityDescription,
+    GitHubSensorInformationEntityDescriptionMixin,
+):
+    """Describes GitHub informatio sensor entity."""
+
+
+@dataclass
+class GitHubSensorIssueEntityDescription(
+    GitHubSensorBaseEntityDescription,
+    GitHubSensorIssueEntityDescriptionMixin,
+):
+    """Describes GitHub issue sensor entity."""
+
+
+SENSOR_DESCRIPTIONS: tuple[
+    GitHubSensorInformationEntityDescription | GitHubSensorIssueEntityDescription,
+    ...,
+] = (
+    GitHubSensorInformationEntityDescription(
+        key="stargazers_count",
+        name="Stars",
+        icon="mdi:star",
+        native_unit_of_measurement="Stars",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda data: data.stargazers_count,
+        coordinator_key="information",
+    ),
+    GitHubSensorInformationEntityDescription(
+        key="subscribers_count",
+        name="Watchers",
+        icon="mdi:glasses",
+        native_unit_of_measurement="Watchers",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        state_class=SensorStateClass.MEASUREMENT,
+        # The API returns a watcher_count, but subscribers_count is more accurate
+        value_fn=lambda data: data.subscribers_count,
+        coordinator_key="information",
+    ),
+    GitHubSensorInformationEntityDescription(
+        key="forks_count",
+        name="Forks",
+        icon="mdi:source-fork",
+        native_unit_of_measurement="Forks",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda data: data.forks_count,
+        coordinator_key="information",
+    ),
+    GitHubSensorInformationEntityDescription(
```

Comment:
Why would anyone be interested in this one?

Reply:
You right, no one cares